geostore/permissions.py

```diff
@@ -0,0 +1,34 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class LayerPermission(BasePermission):
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+        else:
+            return request.user.has_perm('geostore.can_manage_layers')
+
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_superuser:
+            return True
+
+        has_perm = request.user.has_perm('geostore.can_manage_layers')
+
+        if obj.authorized_groups.count():
```

Comment:
don't use count if you plan to iterate on qs

https://docs.djangoproject.com/en/2.2/topics/db/optimization/#don-t-overuse-count-and-exists

Reply:
In fact we use it only if it contains something, exists() is a better way.